android/KMEA/app/src/main/java/com/tavultesoft/kmea/packages/PackageProcessor.java

```diff
@@ -270,18 +275,35 @@ public static Keyboard getKeyboard(JSONObject json, String packageID, int index)
   }
 
   /**
-   * Parse a kmp.json JSON object and return the keyboard count. Lexical model packages return 0
+   * Parse a kmp.json JSON object and return the number of touch-layout keyboards (.JS) matching
+   * keyboard IDs in the "keyboards" JSONArrray.
+   * Lexical model packages return 0
    * @param json kmp.json as a JSON object
-   * @return int of number of keyboards. 0 if not found
+   * @return int of number of matching touch-layout keyboards. 0 if not found
    */
   public static int getKeyboardCount(JSONObject json) {
     int count = 0;
     try {
       if (!json.has(PP_KEYBOARDS_KEY)) {
         return count;
       }
+
+      // Count the number of keyboard ID's that have a matching JS file is in the kmp
       JSONArray keyboards = json.getJSONArray(PP_KEYBOARDS_KEY);
-      count = keyboards.length();
+      for (int k = 0; k <  keyboards.length(); k++) {
+        JSONObject keyboardObj = keyboards.getJSONObject(k);
+        String keyboardID = keyboardObj.getString(PP_KEYBOARD_ID_KEY);
+        String expectedKeyboardFilename = keyboardID + FileUtils.JAVASCRIPT;
+        JSONArray files = json.getJSONArray(PP_FILES_KEY);
+        for (int f = 0; f < files.length(); f++) {
+          JSONObject file = files.getJSONObject(f);
+          String filename = file.getString(PP_FILES_NAME_KEY);
+          if (FileUtils.hasJavaScriptExtension(filename) && filename.equals(expectedKeyboardFilename)) {
```

Comment:
```suggestion
          if (filename != null && filename.equals(expectedKeyboardFilename)) {
```

The first test is redundant but the replacement here is probably good for safety?